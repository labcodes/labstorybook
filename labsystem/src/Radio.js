/* eslint-disable react/require-default-props */
import React from "react";
import PropTypes, { string, number, bool } from "prop-types";
import { isUndefined } from "lodash";

// Radio //

export default class Radio extends React.Component {
  static propTypes = {
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([string, number, bool]).isRequired,
    disabled: PropTypes.bool,
    checked: PropTypes.bool,
    defaultChecked: PropTypes.bool,
    className: PropTypes.string,
    onChange: PropTypes.func,
  };

  static defaultProps = {
    disabled: false,
    checked: undefined,
    defaultChecked: undefined,
    className: undefined,
    onChange: () => {},
  };

  constructor(props) {
    super(props);
    const { defaultChecked, checked, id } = props;
    if (!isUndefined(defaultChecked) && !isUndefined(checked)) {
      // eslint-disable-next-line no-console
      console.warn(
        `You are setting both checked and defaultChecked for radio input ${id} at the same time. We always initialize the radio with defaultChecked. Make sure this is the behaviour you want.`
      );
    }

    let localChecked = false;
    if (defaultChecked) {
      localChecked = defaultChecked;
    } else if (!isUndefined(checked)) {
      localChecked = checked;
    }

    this.state = {
      localChecked,
    };
  }

  componentDidUpdate(prevProps) {
    const { checked } = this.props;
    if (checked !== prevProps.checked) {
      // eslint-disable-next-line react/no-did-update-set-state
      this.setState(() => ({ localChecked: checked }));
    }
  }

  handleOnChange = (event) => {
    const { onChange } = this.props;
    if (!isUndefined(onChange)) {
      onChange(event);
    }

    this.setState((state) => ({ localChecked: !state.localChecked }));
  };

  render() {
    const { className, id, label, disabled, checked, name, value } = this.props;

    return (
      <React.Fragment>
        <input
          className={`lab-radio ${className || ""}`}
          type="radio"
          id={id}
          {...(disabled ? { disabled } : undefined)}
          checked={checked}
          name={name}
          value={value}
          onChange={this.handleOnChange}
        />
        <label className="lab-radio__label" htmlFor={id}>
          <span className="lab-radio__container" />
          {label}
        </label>
      </React.Fragment>
    );
  }
}
