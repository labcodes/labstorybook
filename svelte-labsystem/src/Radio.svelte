<script>
  export let id;
  export let name;
  export let label;
  export let group = undefined;
  export let value = undefined;
  export let disabled = undefined;

  let className = "";
  export { className as class };
</script>

<input
  class={`lab-radio${` ${className}` || ''}`}
  type="radio"
  {id}
  {name}
  {value}
  {disabled}
  bind:group
  on:change />

<label class="lab-radio__label" for={id}>
  <span class="lab-radio__container" />
  {label}
</label>
