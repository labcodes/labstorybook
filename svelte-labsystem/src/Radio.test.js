import { render } from "@testing-library/svelte";
import Radio from "./Radio.svelte";

describe("Radio", () => {
  it("renders with base props", async () => {
    const { container } = render(Radio, {
      id: "test",
      name: "test",
      label: "Test",
    });
    expect(container).toMatchSnapshot();
  });

  it("renders as expected when passing disabled as true", async () => {
    const { container, getByLabelText } = render(Radio, {
      id: "test-input",
      name: "test-input",
      label: "Test",
      disabled: true,
    });

    expect(container).toMatchSnapshot();
    expect(getByLabelText("Test").disabled).toBeTruthy();
  });

  it("renders as expected when passing a different className", async () => {
    const { container } = render(Radio, {
      id: "test",
      name: "test",
      label: "Test",
      class: "dummy-class",
    });
    expect(container).toMatchSnapshot();
  });
});
