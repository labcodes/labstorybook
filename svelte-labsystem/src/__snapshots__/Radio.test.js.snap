// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radio renders as expected when passing a different className 1`] = `
<body>
  <div>
    <input
      class="lab-radio dummy-class"
      id="test"
      name="test"
      type="radio"
      value="undefined"
    />
     
    <label
      class="lab-radio__label"
      for="test"
    >
      <span
        class="lab-radio__container"
      />
       
      Test
    </label>
  </div>
</body>
`;

exports[`Radio renders as expected when passing disabled as true 1`] = `
<body>
  <div>
    <input
      class="lab-radio "
      disabled=""
      id="test-input"
      name="test-input"
      type="radio"
      value="undefined"
    />
     
    <label
      class="lab-radio__label"
      for="test-input"
    >
      <span
        class="lab-radio__container"
      />
       
      Test
    </label>
  </div>
</body>
`;

exports[`Radio renders with base props 1`] = `
<body>
  <div>
    <input
      class="lab-radio "
      id="test"
      name="test"
      type="radio"
      value="undefined"
    />
     
    <label
      class="lab-radio__label"
      for="test"
    >
      <span
        class="lab-radio__container"
      />
       
      Test
    </label>
  </div>
</body>
`;
